source/Ship.cpp

```diff
@@ -1215,6 +1215,7 @@ void Ship::Move(vector<Visual> &visuals, list<shared_ptr<Flotsam>> &flotsam)
 			// Have all ships exit hyperspace at the same distance so that
 			// your escorts always stay with you.
 			double distance = (HYPER_C * HYPER_C) * .5 * HYPER_A + HYPER_D;
+			distance += currentSystem->ArrivalDistance();
```

Comment:
So one thing to mention is that this is usually not the correct usage of the **`System`** arrival distance, because the `target` position is generally that of a planetary body, and not the system center. This would, however, be (part of) the correct usage of a **`Planet`**'s "HyperspaceExclusionRadius". Only when the player jumps into a system that has no planets with spaceports and has also not explicitly selected the planet in that system as a landing target would `target` be that of the system center (aka `Point { 0, 0 }`).

I guess the question is, "how correct do we want to be?" (Or, "how much vector math would you like to do?")

Reply:
Given the choice, I would prefer the hypserspace exclusion radius to be from system center rather from the planet. How complicated is the vector math involved?

Reply:
A simple solution would be to always use the system center as reference point when `arrivalDistance` is not zero. That should be a one-liner of code.

I didn't check yet how complicated the complex solution would be.

Reply:
I implemented the simple solution with my last commit.